### PR TITLE
CRYP-74: implement backend api for getting a users transactions

### PR DESCRIPTION
**Pull request by alexbara2000**
Added ability to get transactions for a user in the eth architecture

### CODE REVIEW

File: packages/api/src/services/transactions.service.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import { EdgeGatewayStrategyFactory } from "@cryptify/api/src/gateways/edge-gateway/edge_gateway_strategy_factory";
+import { CurrencyType } from "@cryptify/common/src/domain/currency_type";
+import { Transaction } from "@cryptify/common/src/domain/entities/transaction";
+
+@Injectable()
+export class TransactionsService {
+    constructor(private readonly edgeGatewayStrategyFactory: EdgeGatewayStrategyFactory) {}
+
+    async findAll(req: GetTransactionsRequest): Promise<Transaction[]> {
+        // Map through the currency types to build a list of strategies, then map through each strategy
```
Comment (jason-gerard) [inline packages/api/src/services/transactions.service.ts:12]: Delete comment

File: packages/common/src/requests/get_transaction_request.ts [hunk 2]
```diff
@@ -0,0 +1,4 @@
+import { InferType } from "yup";
+import { getTransactionsSchema } from "../validations/get_transactions_schema";
+
+export type GetTransactionsRequest = InferType<typeof getTransactionsSchema>;
```
Comment (jason-gerard) [inline packages/common/src/requests/get_transaction_request.ts:4]: should be an interface

Comment (jason-gerard) [reply packages/common/src/requests/get_transaction_request.ts:4]: Nevermind, I guessed none of the other ones were changed

File: packages/eth-edge/src/services/transactions.service.ts [hunk 3]
```diff
@@ -52,4 +55,12 @@ export class TransactionsService {
         // has already been processed by the system
         await this.transactionsRepository.save(transactions);
     }
+
+    async findAll(userId: number): Promise<Transaction[]> {
+        const userWallets = await this.walletsService.findAll(userId);
+        const addresses = userWallets.map((t) => t.address.toLowerCase());
```
Comment (jason-gerard) [inline packages/eth-edge/src/services/transactions.service.ts:61]: We should make sure to also lower case all addresses in general when we insert?
```suggestion
        const addresses = userWallets.map((transaction) => transaction.address.toLowerCase());
```

File: packages/api/src/services/transactions.service.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+        // and get the transactions from the associated edge service in parallel, finally flatten the 2D
+        // array of wallets into a single array
```
Comment (jason-gerard) [inline packages/api/src/services/transactions.service.ts:14]: Small typo in comment

```suggestion
        // Map through the currency types to build a list of strategies, then map through each strategy
        // and get the transactions from the associated edge service in parallel, finally flatten the 2D
        // array of transactions into a single array
```

File: packages/eth-edge/src/controllers/transactions.controller.ts [hunk 5]
```diff
@@ -13,4 +17,10 @@ export class TransactionsController {
         this.transactionsService.handleAddressActivityEvent(body);
         return;
     }
+
+    @Get("users/:id/transactions")
+    async findAll(@Param() params: GetTransactionsRequest): Promise<Transaction[]> {
+        const getTransactionReq = await useValidate(getTransactionsSchema, params);
+        return this.transactionsService.findAll(getTransactionReq.id);
```
Comment (jason-gerard) [inline packages/eth-edge/src/controllers/transactions.controller.ts:24]: Ideally the services would take the request object like what is done in the api but i noticed the other `findAll` are done like this so its fine for now

```suggestion
        return this.transactionsService.findAll(getTransactionReq);
```

File: packages/eth-edge/src/services/transactions.service.ts [hunk 6]
```diff
@@ -52,4 +55,12 @@ export class TransactionsService {
         // has already been processed by the system
         await this.transactionsRepository.save(transactions);
     }
+
+    async findAll(userId: number): Promise<Transaction[]> {
+        const userWallets = await this.walletsService.findAll(userId);
```
Comment (jason-gerard) [inline packages/eth-edge/src/services/transactions.service.ts:60]:
```suggestion
        const wallets = await this.walletsService.findAll(userId);
```

**Review by jason-gerard — APPROVED** [review]

File: packages/common/src/requests/get_transaction_request.ts (hunk 2 above)
Comment (alexbara2000) [reply packages/common/src/requests/get_transaction_request.ts:4]: get_wallet_request is not an interface, should I change both to be interfaces?

File: packages/client/gateways/transactions_gateway.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
+import { JwtToken } from "@cryptify/common/src/domain/jwt_token";
+import { GetTransactionsRequest } from "@cryptify/common/src/requests/get_transaction_request";
+import { AbstractApiGateway } from "./abstract_api_gateway";
+import { Method } from "@cryptify/common/src/utils/gateway/abstract_gateway";
+import { Transaction } from "@cryptify/common/src/domain/entities/transaction";
+
+export class TransactionsGateway extends AbstractApiGateway {
```
Comment (jason-gerard) [inline packages/client/gateways/transactions_gateway.ts:7]: Not sure this belongs in this pull request

**Pull request merged**